pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -155,6 +196,12 @@ type reconcilePhase struct {
 // queue.
 func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result reconcile.Result, err error) {
 
+	counter, err := reconcileCounters.GetMetricWithLabelValues(request.Name)
```

Comment:
`GetMetricWithLabelValues()` returns an error only if the number of label values doesn't match with the description of the metric. In general you want to fail hard as it's a sign that the code using the metric doesn't reflect the metric definition. You could use `WithLabelValues()` that would panic where `GetMetricWithLabelValues()` would have returned an error.

```
reconcileCounters.WithLabelValues(request.Name).Inc()
```